ISISScientificComputing/autoreduce
# Split up PostProcessAdmin

@dtasev as promised. Probably best not for this release but to test in between on dev env

### Summary of work

#### What is done:

- Introduced classes as per discussion with @dtasev and @rickybassom 
  - Due to #875 `TemporaryReductionDirectory` no longer extends `ReductionDirectory` and instead makes use of the `tempfile` module and is actually nicer as we can completely delegate responsibility of cleanup of temp dirs (Currently broken on dev)
- Introduced new exceptions within `reduction_exceptions.py` and moved the existing `SkippedRunException`
- Move reduction responsibility away from `PostProcessAdmin` to `reduction_service.py`
- More testing
- Remove explicit validation as it becomes inherent within design.

#### What is working:
Through limited testing on dev
- Runs successfully for some flat output and non flat instruments GEM/WISH/HRPD
- With logging as is reduction_log modal is still populated
- Data location is still populated

#### What needs to be done:
- Improve the way logging is setup in both `post_process_admin` and `reduction_service`
- Look at `channels_redirected` (Out of scope of this PR potentially)
- Rename `post_process_admin` as discussed (Again potentially out of scope.)
- `reduction_service` should really be in a package separate to `queue_processors`
- The reduce method within `post_process_admin` remains untested


### How to test your work
Deploy the branch to dev and check runs go through, including logs written to and displayed in WebApp

### Additional comments
 

Closes #756 Closes #865


**Before merging ensure the release notes have been updated**
